src/Pos/Types/Arbitrary.hs

```diff
@@ -22,226 +22,53 @@ module Pos.Types.Arbitrary
-                                            makeScriptAddress)
-import           Pos.Core.Coin             (coinToInteger, divCoin, unsafeSubCoin)
-import           Pos.Core.Types            (BlockVersion (..), Coin, SoftwareVersion (..),
-                                            applicationNameMaxLength, mkApplicationName)
-import qualified Pos.Core.Types            as Types
-import           Pos.Crypto                (Hash, PublicKey, SecretKey, Share,
-                                            SignTag (SignTxIn), hash, sign, toPublic)
+import           Pos.Core.Address          (makePubKeyAddress)
+import           Pos.Core.Arbitrary        (CoinPairOverflowMul (..),
+                                            CoinPairOverflowSub (..),
+                                            CoinPairOverflowSum (..),
+                                            DoubleInZeroToOneRange (..),
+                                            IntegerToCoinNoOverflow (..),
+                                            IntegerToCoinOverflow (..),
+                                            LessThanZeroOrMoreThanOne (..),
+                                            SafeCoinPairMul (..), SafeCoinPairSub (..),
+                                            SafeCoinPairSum (..), SafeWord (..),
+                                            SmallHashMap (..))
+import           Pos.Core.Types            (Coin)
+import           Pos.Crypto                (Hash, SecretKey, SignTag (SignTxIn), hash,
+                                            sign, toPublic)
 import           Pos.Crypto.Arbitrary      ()
 import           Pos.Data.Attributes       (mkAttributes)
 import           Pos.Merkle                (MerkleRoot (..), MerkleTree, mkMerkleTree)
-import           Pos.Script                (Script)
-import           Pos.Script.Examples       (badIntRedeemer, goodIntRedeemer, intValidator)
 import           Pos.Txp.Core.Types        (Tx (..), TxAux (..), TxDistribution (..),
                                             TxIn (..), TxInWitness (..), TxOut (..),
                                             TxOutAux (..), TxProof (..), TxSigData (..),
                                             mkTx)
-import           Pos.Types.Arbitrary.Core  ()
 import           Pos.Util                  (makeSmall)
 
 ----------------------------------------------------------------------------
 -- Arbitrary core types
 ----------------------------------------------------------------------------
 
-instance Arbitrary Script where
-    arbitrary = elements
-        [intValidator, goodIntRedeemer, badIntRedeemer]
-
-instance Arbitrary Types.Address where
-    arbitrary = oneof [
-        makePubKeyAddress <$> arbitrary,
-        makeScriptAddress <$> arbitrary,
-        makeRedeemAddress <$> arbitrary,
-        Types.UnknownAddressType <$> choose (3, 255) <*> scale (min 150) arbitrary
-        ]
-
-deriving instance Arbitrary Types.ChainDifficulty
+-- It conflicts with core arbitrary
```

Comment:
If i delete the file, branch won't compile, because these types are needed in tests.
???